tests/CMakeLists.txt

```diff
@@ -243,127 +252,147 @@ FOREACH(_test ${_tests})
+	DEPENDS ${ASPECT_BINARY} ${CMAKE_CURRENT_BINARY_DIR}/${testname}.x.prm ${_testdepends})
     ELSE()
-      ADD_CUSTOM_COMMAND(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/screen-output
-	COMMAND mpirun -np ${_mpi_count} ${ASPECT_BINARY} ${CMAKE_CURRENT_BINARY_DIR}/${_test}.x.prm
-		> ${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/screen-output.tmp
-	COMMAND ${_replacement_script} screen-output <${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/screen-output.tmp >${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/screen-output
-	DEPENDS ${ASPECT_BINARY}  ${CMAKE_CURRENT_BINARY_DIR}/${_test}.x.prm ${_testdepends}
-	)
+      ADD_CUSTOM_COMMAND(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/output-${testname}/screen-output
+	COMMAND mpirun -np ${_mpi_count} ${ASPECT_BINARY} ${CMAKE_CURRENT_BINARY_DIR}/${testname}.x.prm
+		> ${CMAKE_CURRENT_BINARY_DIR}/output-${testname}/screen-output.tmp
+	COMMAND ${_replacement_script} screen-output <${CMAKE_CURRENT_BINARY_DIR}/output-${testname}/screen-output.tmp >${CMAKE_CURRENT_BINARY_DIR}/output-${testname}/screen-output
+	DEPENDS ${ASPECT_BINARY} ${CMAKE_CURRENT_BINARY_DIR}/${testname}.x.prm ${_testdepends})
     ENDIF()
 
-    # command to normalize all output files and all reference output files
-    ADD_CUSTOM_TARGET(${_test}.normalize
-	COMMAND ${PERL_EXECUTABLE} -pi
+
+    GET_FILENAME_COMPONENT(ASPECT_BASE_DIR ${CMAKE_CURRENT_SOURCE_DIR} PATH)
+
+    # commands to normalize output
+    FOREACH(_output_name ${_output_files})
+	ADD_CUSTOM_COMMAND(OUTPUT ${_output_name}.notime
+	  COMMAND ${PERL_EXECUTABLE}
 		  ${CMAKE_CURRENT_SOURCE_DIR}/normalize.pl
-		  `find ${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/ -type f`
-	DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/screen-output)
+		  ${_output_name} ${ASPECT_BASE_DIR}
+		  >${_output_name}.notime
+          DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/output-${testname}/screen-output)
+    ENDFOREACH()
+
+    # commands to create and normalize reference output
+    FOREACH(_name ${_relative_output_filenames})
+	ADD_CUSTOM_COMMAND(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/output-${testname}/${_name}.cmp.notime
+	  COMMAND ${PERL_EXECUTABLE}
+		  ${CMAKE_CURRENT_SOURCE_DIR}/normalize.pl
+		  ${CMAKE_CURRENT_SOURCE_DIR}/${testname}/${_name} ${ASPECT_BASE_DIR}
+		  >${CMAKE_CURRENT_BINARY_DIR}/output-${testname}/${_name}.cmp.notime
+          DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${testname}/${_name})
+    ENDFOREACH()
+
+    FOREACH(_name ${_relative_output_filenames})
+      SET(_ref_file ${CMAKE_CURRENT_BINARY_DIR}/output-${testname}/${_name}.cmp.notime)
+      SET(_run_file ${CMAKE_CURRENT_BINARY_DIR}/output-${testname}/${_name}.notime)
+
+      ADD_CUSTOM_COMMAND(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/output-${testname}/${_name}.diff
+	COMMAND
+	    ${CMAKE_CURRENT_SOURCE_DIR}/cmake/diff_test.sh 
+		${TEST_DIFF}
+		${testname}/${_name}
+		${CMAKE_CURRENT_SOURCE_DIR}
+		${CMAKE_CURRENT_BINARY_DIR}
+	DEPENDS ${_ref_file}
+	        ${_run_file})
```

Comment:
I thought this was the problem with `BYPRODUCT`: `${_run_file}` is an unsatisfiable dependency because we have no targets that produce them. Don't you want to depend on `screen-output.notime` here for which there actually *is* a rule that has it as output?